submodules/misc/misc.js

```diff
@@ -402,6 +402,66 @@ define(function(require) {
 						}
 					}
 				},
+				'set_alert_info[]': {
+					name: self.i18n.active().oldCallflows.set_ringtone,
+					icon: 'play',
+					category: self.i18n.active().oldCallflows.advanced_cat,
+					module: 'set_alert_info',
+					tip: self.i18n.active().oldCallflows.set_ringtone_tip,
+					data: {
+						alert_info: ''
+					},
+					rules: [
+						{
+							type: 'quantity',
+							maxSize: '1'
+						}
+					],
+					isUsable: 'true',
+					weight: 20,
+
+					caption: function(node, caption_map) {
+						return (node.getMetadata('alert_info') || '');
+					},
+
+					edit: function(node, callback) {
+						var popup_html = $(self.getTemplate({
+								name: 'set-ringtone',
+								data: {
+									data_cid: {
+										'alert_info': node.getMetadata('alert_info') || ''
+									}
+								},
+								submodule: 'misc'
+							})),
+							popup;
+
+						$('#add', popup_html).click(function() {
+							var alert_info_val = $('#alert_info', popup_html).val();
+
+							node.setMetadata('alert_info', alert_info_val);
+
+							node.caption = alert_info_val;
+
+							popup.dialog('close');
+						});
+
+						popup = monster.ui.dialog(popup_html, {
+							title: self.i18n.active().oldCallflows.set_ringtone_title,
+							beforeClose: function() {
+								if (typeof callback === 'function') {
+									callback();
+								}
+							}
+						});
+
+						monster.ui.tooltips(popup);
+
+						if (typeof callback === 'function') {
+							callback();
+						}
```

Comment:
This callback is used to re-render the flow, we only need to call it before the popup is closed ([line](https://github.com/2600hz/monster-ui-callflows/pull/49/files#diff-5a2dc5d447c4aaf156c44527951a6466R451)), not when the popup is opened.

You can get ride of this block as well.